plugins/rsync_server.py

```diff
@@ -0,0 +1,157 @@
+"""
+Plugin to run rsync to sync files to a server.
+"""
+
+# Local imports
+import brokkr.pipeline.base
+import brokkr.pipeline.decode
+
+# Standard library imports
+import sys
+import subprocess
+import os.path
+
+
+class RsyncServer(brokkr.pipeline.base.OutputStep):
+    """Sync files to a server via rsync"""
+
+    def __init__(self,
+                 server,
+                 local_path,
+                 server_path,
+                 update_time,
+                 include=None,
+                 time_key='rsync_time',
+                 username='pi',
+                 **output_step_kwargs):
+        """
+        Sync files to a server via rsync.
+
+        This class will "inject" a new entry into the dictionary Brokkr passes
+        in pipelines that tracks the last time the rsync was performed.
+
+        If you want to rsync two different paths in the same pipeline, you
+        would provide two different steps and unique `time_key`s for each step.
+
+        Parameters
+        ----------
+        server : str
+            The server to sync the files to.
+        local_path : str
+            The local path to the files to be synced.
+        server_path : str
+            The path on the server the files will be synced to
+        username : str
+            The user name on the server.
+        update_time : int
+            The number of seconds that must elapse between successful syncs.
+            Sets the cadence of syncing to the server.
+        include : str
+            Typically, a pattern of files to be synced. Passed to the `include` option
+            of rsync.
+        time_key : str
+            The key corresponding to the time to track the last update.
+        output_step_kwargs : **kwargs, optional
+            Keyword arguments to pass to the OutputStep constructor.
+
+        """
+        # Pass arguments to superclass init
+        super().__init__(**output_step_kwargs)
+
+        self._previous_data = None
+
+        self.server = server
+        self.local_path = local_path
```

Comment:
```suggestion
        self.local_path = brokkr.utils.output.render_output_filename(
            output_path=local_path,
            filename_template="")
```

Automatically resolve a relative path to be relative to the system data directory, handle variable substantiation, etc.